cmd/node/node.go

```diff
@@ -348,6 +352,25 @@ func (app SpacemeshApp) stopServices() {
 
 }
 
+func getEdIdentity() *signing.EdSigner {
+	// read key from file if exist
+	log.Info("Reading identity from file")
+	dataDir, err := filesystem.GetSpacemeshDataDirectoryPath()
+	if err != nil {
+		log.Error("Could not get data path err=%v", err)
+		return signing.NewEdSigner()
+	}
+
+	buff, e1 := ioutil.ReadFile( dataDir + "/identity.ed")
+	edSgn, e2 := signing.NewEdSignerFromBuffer(buff)
+	if e1 != nil || e2 != nil {
```

Comment:
You should differentiate between "real errors" (i.e. problems access the disk) and "unknown file" kind of errors. "Unknown file" should lead to creating a new sig, any other error should be reported back to the caller

Reply:
Also, when creating new sign you should write it to a file

Reply:
Fixed